README.md

```diff
@@ -1,7 +1,100 @@
-Generate a compatibility matrix for plugins against Jenkins core.
+Plugin Compatibility Tester (PCT)
+------
+
+Generates a compatibility matrix for plugins against Jenkins core.
 
 See https://wiki.jenkins-ci.org/display/JENKINS/Plugin+Compatibility+Tester for background.
 
+## Running PCT 
+
+### Running PCT in Docker
+
+It is recommended to run PCT in the prepared Docker image.
+You can find it [here](hub.docker.com/r/jenkins/pct/).
+
+#### Examples
+
+This command will clone of a repository and run PCT against the latest Jenkins Core:
+
+```shell
+docker run --rm -v maven-repo:/root/.m2 -v $(pwd)/out:/pct/out -e ARTIFACT_ID=ssh-slaves -e VERSION=ssh-slaves-1.24 jenkins/pct
+```
+
+This command will run PCT for the latest plugin version against the specified Jenkins WAR.
+PCT supports running against SNAPSHOT builds, but PCT will have to install local Maven artifacts in such case.
+
+```shell
+docker run --rm -v maven-repo:/root/.m2 -v $(pwd)/out:/pct/out -v my/jenkins.war:/pct/jenkins.war:ro -e ARTIFACT_ID=ssh-slaves jenkins/pct
+```
+
+This command will run PCT against custom versions of Jenkins and the plugin specified by volumes.
+
+```shell
+docker run --rm -v maven-repo:/root/.m2 -v $(pwd)/out:/pct/out -v my/jenkins.war:/pct/jenkins.war:ro -v my/plugin:/pct/plugin-src:ro jenkins/pct
+```
+
+The command below will run PCT for a branch in a custom repository.
+
+```shell
+docker run --rm -v maven-repo:/root/.m2 -v $(pwd)/out:/pct/out -e CHECKOUT_SRC=https://github.com/oleg-nenashev/job-restrictions-plugin.git -e VERSION=JENKINS-26374 jenkins/pct
+```
+
+#### Configuration
+
+Environment variables:
+
+* `ARTIFACT_ID` - ID of the artifact to be tested.
+The image will be able to determine this ID automatically if `CHECKOUT_SRC` or `/pct/plugin-src` are defined.
+* `VERSION` - tag/commit/branch to be checked out and tested. `master` by default
```

Comment:
Confusing name. I would expect this to be a Maven version. But you seem to be referring to a Git ref name?

Reply:
Yes, I can change that

Reply:
Created https://issues.jenkins-ci.org/browse/JENKINS-50787 as a follow-up